CONTRIBUTING.md

```diff
@@ -81,6 +81,27 @@ reformats the code (based on configuration in the `node_modules/resin-lint/confi
 file). Beyond that, we have a preference for Javascript promises over callbacks, and for
 `async/await` over `.then()`.
 
+## Updating upstream dependencies
+
+In order to get proper nested changelogs, when updating upstream modules that are in the repo.yml
+(like the balena-sdk), the commit body has to contain a line with the following format:
+```
+Update balena-sdk from 12.0.0 to 12.1.0
+```
+
+Since this is error prone, it's suggested to use the following npm script:
+```
+npm run update balena-sdk ^12.1.0
+```
+This will create a new branch (only if you are currently on master), run `npm update` with the
+version you provided as a target and commit the package.json & npm-shrinkwrap.lock. The script by
```

Comment:
`npm-shrinkwrap.json` ? (lock vs json)

Reply:
`npm-shrinkwrap.json` gets published to NPM, so the deps will be set on package download, `package-lock.json` never gets published. It's something that's bugged me for a while (there are reasons for it, npm-shrinkwrap was the legacy way of doing this and also respects all deps in the lock file, not just top-level; it also means that you get exactly what you specify when your NPM package is downloaded as a module, instead of just what's compatible with `package.json`).

Reply:
hmm I meant the typo: `npm-shrinkwrap.lock` _vs._ `npm-shrinkwrap.json`